### DISCUSSION TITLE
/articles/microsoft-graph-webhook-delta-query/

### DISCUSSION THREAD

**Discussion by giscus** [Voitanos/voitanos.io-comments#237] [Comments]
Learn about a feature that’s been in Microsoft Graph for a very long time. But it’s always surprising to me how many people aren’t aware of it.
https://www.andrewconnell.com/articles/microsoft-graph-webhook-delta-query/

**Comment by sinnitesh** [comment]
insightfull post, loved reading it.

**Comment by simolazza** [comment]
Hello, 
thanks for the post! wanted to know, is it possible to use webhook on a sharepoint list also to trigger the change of permissions on a document?

Thanks

**Reply by andrewconnell** [reply]
Not sure... that's something you'd have to test. I'm not sure changing the role assignment (how permissions are done) would trigger the update event that would trigger a webhook. The role assignment is separate from the actual list item as it's a separate collection with three elements: the item ID, the user ID, and the permission.

If you test it and find out, be sure to come back and let us know!

**Comment by icelava** [comment]
I'm trying to sustain subscriptions to Teams channel messages on the reauthorizationRequired event. But whether on Power Automate with HTTP With Entra ID connector or Graph Explorer, I'm not allowed to PATCH subscriptions.

```
{
  "error": {
    "code": "ExtensionError",
    "message": "Operation: Update; Exception: [Status Code: Forbidden; Reason: 
                           Writes are not allowed for the specified subscription with Id 
                           'DIFFERENT_GUID_FROM_SUBSCRIPTION' by the caller]",
    "innerError": {
      "date": "2025-03-24T14:06:10",
      "request-id": "requestId",
      "client-request-id": "clientRequestId"
    }
  }
}
```

**Reply by andrewconnell** [reply]
@icelava I haven't run into this before, but to me, the error indicates you've obtained an access token from an Entra ID tenant and you're trying to update the Graph webhook subscription in another M365 tenant. In other words, you've obtained the access token via an Entra ID app for use with MS Graph from one tenant, but you're trying to use it in another tenant where that app hasn't been granted access.

If you can share more, that would help troubleshoot and better understand it. For instance:

- share the decoded JWT access token (you can remove claims like the `oid`, `upn`, `name`, the header & the signature so the token won't be useful to anyone else... I'd specifically want to see the claims `tid`, `aud`, `scp`, `iss`... for GUIDs, feel free to XXX-out the last few segments... need to see at least the first one or two segments)... *if you're concerned what I'm asking for, you can rest assured as the things I've said you can remove will take the PII out of the request, and the claims I'm saying I want to see will help troubleshoot... by only sharing SOME of the claims & omitting the digital signature, the token will be useless to others and thus, not compromise you*)
- share the HTTP request (the header & payload
- share the HTTP response (the full header & payload if there's more than what you posted above

> *DISCLAIMER: I formatted your response by fencing the HTTP response to improve readability. Please do this in your reply as well to make the response more readable. TIA!*

**Reply by icelava** [reply]
But I only have one tenant - my company.

From the Graph Explorer JWT
iss = https://sts.windows.net/MY_TENANT_ID/
tid = MY_TENANT_ID
aud = 00000003-0000-0000-c000-000000000000
scp = Application.Read.All Channel.ReadBasic.All ChannelMessage.Read.All ChannelSettings.Read.All Chat.Read Chat.ReadBasic Contacts.ReadWrite DeviceManagementRBAC.Read.All DeviceManagementServiceConfig.Read.All Files.ReadWrite.All Group.ReadWrite.All IdentityRiskEvent.Read.All Mail.Read Mail.ReadWrite MailboxSettings.ReadWrite Notes.ReadWrite.All openid People.Read Place.Read Presence.Read Presence.Read.All PrinterShare.ReadBasic.All PrintJob.Create PrintJob.ReadBasic profile Reports.Read.All Sites.ReadWrite.All Tasks.ReadWrite User.Read User.ReadBasic.All User.ReadWrite User.ReadWrite.All email Subscription.Read.All

I have no idea how to see the JWT from Power Automate.

The PATCH request as documented is set Content-type to application/json
```
{
   "expirationDateTime":"2025-03-25T10:30:45.9356913Z"
}
```

The forbidden response in Graph Explorer is exactly as pasted above. From Power Automate HTTP with Entra ID action

```
{
    "statusCode": 403,
    "headers": {
        "Cache-Control": "no-cache",
        "Transfer-Encoding": "chunked",
        "Vary": "Accept-Encoding",
        "Strict-Transport-Security": "max-age=31536000",
        "request-id": "REQUEST_ID",
        "client-request-id": "REQUEST_ID",
        "x-ms-ags-diagnostic": "{\"ServerInfo\":{\"DataCenter\":\"Southeast Asia\",\"Slice\":\"E\",\"Ring\":\"5\",\"ScaleUnit\":\"000\",\"RoleInstance\":\"INSTANCE)ID\"}}",
        "x-ms-environment-id": "ENVIRONMENT_ID",
        "x-ms-tenant-id": "TENANT_ID",
        "x-ms-dlp-re": "-|-",
        "x-ms-dlp-gu": "-|-",
        "x-ms-dlp-ef": "-|-/-|-|-",
        "Timing-Allow-Origin": "*",
        "x-ms-apihub-cached-response": "false",
        "x-ms-apihub-obo": "false",
        "Date": "Wed, 26 Mar 2025 08:56:00 GMT",
        "Content-Type": "application/json",
        "Content-Length": "446"
    },
    "body": {
        "error": {
            "code": "ExtensionError",
            "message": "Operation: Update; Exception: [Status Code: Forbidden; Reason: Writes are not allowed for the specified subscription with Id 'NOT_SUBSCRIPTION_ID' by the caller]",
            "innerError": {
                "date": "2025-03-26T08:56:01",
                "request-id": "REQUEST_ID",
                "client-request-id": "REQUEST_ID"
            }
        }
    }
}
```

**Reply by icelava** [reply]
More details on the scenario I am developing for
https://learn.microsoft.com/en-us/answers/questions/2236135/subscribe-to-teams-messages-likes-reacts

**Reply by icelava** [reply]
Additional info: I check Entra Conditional Access and there appears to be no blockages for Power Automate.

```
27/03/2025, 17:12:43.630
icelava@company.co
PowerPlatform-webcontentsv2-Connector
success
```

**Reply by andrewconnell** [reply]
Unfortunately, I'm not sure how you can see the access token that Power Automate is using. From what you've shared, that sounds like the root of the issue... probably need to figure how to get that to inspect what it's using and why you're getting the forbidden response. I'm not sure how to do that as I don't use Power Automate.

The stuff you shared from the Graph Explorer isn't much help because that's only showing you the permissions you've granted to the Graph Explorer which isn't what's being used in this case.

**Reply by icelava** [reply]
I feel if I can at least get it to work via Graph Explorer then I'd be one step closer to figuring out the problem on Power Automate. I've logged a support request with Microsoft to see what other undocumented info they have to offer.

**Reply by andrewconnell** [reply]
Please share what you're seeing using JUST the Graph Explorer. The Power Automate part of your follow-up posts to your question is a black box, making it hard to help with that part. If you're having issues with Graph Explorer, those will be easier to troubleshoot as the full request/response is in the clear. [Renewing the subscription](https://learn.microsoft.com/graph/change-notifications-delivery-webhooks?tabs=http#renew-a-subscription) is pretty straightforward so...

**Reply by icelava** [reply]
Oh my goodness found out the core problem.

On learning Graph API, I was extending the tutorial C# code base to test the subscriptions which is its own app in Entra ID. I then subsequently tested the renew PATCH request from Power Automate (Entra ID connector) and Graph Explorer which turns out them being their own apps are not allowed to modify the subscriptions of the tutorial app even though they use my delegated credentials and granted the same permissions.

So in order for Power Automate to renew subscriptions, they must have been originally subscribed from Power Automate. Vice versa for Graph Explorer.

The documentation doesn’t mention such restrictions.

**Reply by andrewconnell** [reply]
Thanks for following up with your findings!

**Reply by icelava** [reply]
Thank you too.
